coc/http.py

```diff
@@ -67,71 +68,62 @@ def __init__(self, method, path, kwargs: dict=None, api_page=False):
 
 
 class HTTPClient:
-    def __init__(self, client, loop, tokens=None, *, email, password, update_tokens=False):
+    def __init__(self, client, loop, email, password,
+                 key_names, key_count):
         self.client = client
         self.loop = loop
-        self.__session = None
+        self.__session = aiohttp.ClientSession(loop=self.loop)
         self.email = email
         self.password = password
-        self.update_tokens = update_tokens
+        self.key_names = key_names
+        self.key_count = key_count
 
-        asyncio.ensure_future(self.login(tokens))
+        loop.run_until_complete(self.get_keys())
```

Comment:
the whole loop.run_until_complete() was what was bugging out tuba's error with reloading client in same script.....while I dont know what the downside of using ensure_future is (apart from not being able to return anything), afaik it was working

Reply:
Hmm I think I change that line because of not knowing you used `self.request` to access the api/key portal too I think I can safely change that back my bad